lychee-lib/src/collector.rs

```diff
@@ -92,28 +92,39 @@ impl Collector {
             .flatten()
     }
 
+    /// Convenience method to fetch all unique links from inputs
+    /// with the default extensions.
+    pub fn collect_links(self, inputs: Vec<Input>) -> impl Stream<Item = Result<Request>> {
+        self.collect_links_with_ext(inputs, crate::types::FileType::default_extensions())
+    }
+
     /// Fetch all unique links from inputs
     /// All relative URLs get prefixed with `base` (if given).
     /// (This can be a directory or a base URL)
     ///
     /// # Errors
     ///
     /// Will return `Err` if links cannot be extracted from an input
-    pub fn collect_links(self, inputs: Vec<Input>) -> impl Stream<Item = Result<Request>> {
+    pub fn collect_links_with_ext(
+        self,
+        inputs: Vec<Input>,
+        extensions: Vec<String>,
```

Comment:
Can we rename `extensions` to `file_extensions` or wrap `Vec<String>` into a newtype named `FileExtensions`? Or maybe we could use the type that ripgrep uses - the one that is created with `TypesBuilder::new()`?